src/util/simplify_expr_boolean.cpp

```diff
@@ -17,42 +17,45 @@ Author: Daniel Kroening, kroening@kroening.com
 #include "namespace.h"
 #include "std_expr.h"
 
-bool simplify_exprt::simplify_boolean(exprt &expr)
+simplify_exprt::resultt<> simplify_exprt::simplify_boolean(const exprt &expr)
 {
   if(!expr.has_operands())
-    return true;
-
-  exprt::operandst &operands=expr.operands();
+    return unchanged(expr);
 
   if(expr.type().id()!=ID_bool)
-    return true;
+    return unchanged(expr);
 
   if(expr.id()==ID_implies)
   {
-    if(operands.size()!=2 ||
-       operands.front().type().id()!=ID_bool ||
-       operands.back().type().id()!=ID_bool)
-      return true;
+    if(
+      expr.operands().size() != 2 ||
+      expr.operands().front().type().id() != ID_bool ||
+      expr.operands().back().type().id() != ID_bool)
+    {
+      return unchanged(expr);
+    }
 
     // turn a => b into !a || b
 
-    expr.id(ID_or);
-    expr.op0() = boolean_negate(expr.op0());
-    simplify_node(expr.op0());
-    simplify_node(expr);
-    return false;
+    auto new_expr = expr;
+    new_expr.id(ID_or);
+    new_expr.op0() = boolean_negate(new_expr.op0());
+    simplify_node(new_expr.op0());
```

Comment:
Is this actually necessary? We don't consistently do this after `boolean_negate`

Reply:
Indeed, this is a no-op. Added as separate commit to #4874.